docs/design/uptime-reputation-calculation.md

```diff
@@ -0,0 +1,122 @@
+
+1. Any _SN_ (veteran and new ones) requires some time to be offline for maintenance and both should have the same maximum amount of time for it.
+1. The end goal of the uptime checks is to ensure that the network has certain availability and retrievability and uptime check failures affects to them no matter on how long the _SN_ has remained in the network and how much data an _SN_ has, hence node with more or less data must be treated equally.
+
+This design doc is focused on how to design and implement a fair, simple and understandable uptime reputation calculation after our initial approach didn't behave as expected, causing the disqualification of several nodes without having a proof that they have failed enough uptime checks that the _SNs_ were offline long time enough which put in danger the minimum level of the network availability and retrievability.
+
+## Design
+
+The new process for doing the uptime checks is based on once an _SN_ is selected for having the check, if the _SN_ fails the initial check, then it's tracked to periodically receive uptime checks until (a) it successfully passes an uptime check or (b) gets disqualified due to many uptime check failures in one month.
+
+Once an _SN_ has failed the first uptime check, the _SN_ is considered offline and the subtotal time, which the _SN_ is considered offline, is the time elapsed between the first and the last failed uptime check. The total offline time of an _SN_ is the accumulation of the subtotals that the _SN_ has had during the ongoing month and when such total surpasses the maximum permitted percentage time, the _SN_ will get disqualified.
+
+The maximum percentage of time that an _SN_ can be offline will be established with the network commitment in terms of availability and retrievability during one month.
+
+The process is divided in 2 parts:
+
+1. Selecting _SNs_ for receiving uptime checks.
+1. Periodically check the _SNs_ which has failed the first uptime check until they pass one or get disqualified.
+
+For the first part, we foresee to use the current implemented mechanism with some minor modifications for excluding the _SNs_ which are being checked by the second part.
+
+The second part requires a totally new implementation which should be totally independent of any current satellite service for being able in the near future to run it separately for scalability purpose.
+
+
+## Rationale
+
+The approach described in this document present the following advantages:
+
+1. It's clear why an _SN_ get disqualified due to failed uptime checks.
+1. It has a clear proof for an _SNOs_ why their _SNs_ get disqualified due to failed uptime check in case that they request it before the designed time to clean up the historical failed uptime checks data.
+1. It allows to report to _SN_ at any how far is to be disqualified by failing uptime checks.
```

Comment:
Maybe `It allows reporting to a _SN_, at any point, how far it is from disqualified by failing uptime checks.`